dll/win32/browseui/addresseditbox.cpp

```diff
@@ -173,6 +173,21 @@ HRESULT STDMETHODCALLTYPE CAddressEditBox::ParseNow(long paramC)
     return hr;
 }
 
+HRESULT STDMETHODCALLTYPE CAddressEditBox::ShowFileNotFoundError(HRESULT hRet)
+{
+    LPWSTR input;
+    int inputLength = fCombobox.GetWindowTextLength() + 2;
+
+    input = new WCHAR[inputLength];
+    fCombobox.GetWindowText(input, inputLength);
+
+    ShellMessageBoxW(_AtlBaseModule.GetResourceInstance(), fCombobox.m_hWnd, MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TEXT), MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TITLE), MB_OK | MB_ICONERROR, input);
+
+    delete[] input;
+
+    return hRet;
```

Comment:
When using a CComHeapPtr like this, you do not have to worry about cleanup:
```suggestion
    CComHeapPtr<WCHAR> input;
    int inputLength = fCombobox.GetWindowTextLength() + 2;

    input.Allocate(inputLength);
    fCombobox.GetWindowText(input, inputLength);

    ShellMessageBoxW(_AtlBaseModule.GetResourceInstance(), fCombobox.m_hWnd, MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TEXT), MAKEINTRESOURCEW(IDS_PARSE_ADDR_ERR_TITLE), MB_OK | MB_ICONERROR, input);

    return hRet;
```

(lgtm either way, just something to keep in mind :) )

Reply:
I like it!  However, I wasn't able to find `input.GetString()`, but `input.m_pData` worked.  Is that expected?

Reply:
My bad, I typed this in the github interface. (`GetString` is from CAtlString, which you can also use here btw)...

For `CComHeapPtr` (or `CHeapPtr` would also work..) `m_pData` is fine, or you can cast it.
Normally using that is not needed when passing it to functions that accept strings,
but since we are passing it to a vararg function, the compiler cannot know what we want without making it explicit.

Reply:
Sounds good, I'll alter the commit to use CComHeapPtr